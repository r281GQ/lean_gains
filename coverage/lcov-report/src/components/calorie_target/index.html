<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/calorie_target</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/calorie_target
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">47.83% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">47.83% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/23</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="activity_level_selector.jsx"><a href="activity_level_selector.jsx.html">activity_level_selector.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file medium" data-value="bmr_calculation_selector.jsx"><a href="bmr_calculation_selector.jsx.html">bmr_calculation_selector.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file medium" data-value="body_fat_field.jsx"><a href="body_fat_field.jsx.html">body_fat_field.jsx</a></td>
	<td data-value="66.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 66%;"></div><div class="cover-empty" style="width:34%;"></div></div></td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	</tr>

<tr>
	<td class="file medium" data-value="calorie_split_selector.jsx"><a href="calorie_split_selector.jsx.html">calorie_split_selector.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file medium" data-value="custom_calorie_field.jsx"><a href="custom_calorie_field.jsx.html">custom_calorie_field.jsx</a></td>
	<td data-value="66.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 66%;"></div><div class="cover-empty" style="width:34%;"></div></div></td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	</tr>

<tr>
	<td class="file medium" data-value="fat_mathod_selector.jsx"><a href="fat_mathod_selector.jsx.html">fat_mathod_selector.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file low" data-value="fat_selector.jsx"><a href="fat_selector.jsx.html">fat_selector.jsx</a></td>
	<td data-value="28.57" class="pic low"><div class="chart"><div class="cover-fill" style="width: 28%;"></div><div class="cover-empty" style="width:72%;"></div></div></td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="protein_field.jsx"><a href="protein_field.jsx.html">protein_field.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
