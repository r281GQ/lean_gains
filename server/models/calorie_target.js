module.exports = mongoose => {
  const calorieTargetSchema = new mongoose.Schema({
    user: {
      ref: 'User',
      type: mongoose.Schema.Types.ObjectId,
      required: true
    },
    startDate: {
      type: Number
    },
    endDate: {
      type: Number
    },
    isLatest: Boolean,
    rest: {
      calorie: Number,
      protein: Number,
      carbohydrate: Number,
      fat: Number
    },
    training: {
      calorie: Number,
      protein: Number,
      carbohydrate: Number,
      fat: Number
    }
  });

  mongoose.model('CalorieTarget', calorieTargetSchema);
};
