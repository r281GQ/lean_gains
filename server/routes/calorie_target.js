const {
  handleGetCalorieTarget,
  handlePostCalorieTarget
} = require('./../services/calorie_target');
const withAuth = require('./../services/auth_middleware');

module.exports = app => {
  app.get('/api/calorietargets', withAuth, (request, response) => {
    handleGetCalorieTarget(request)
      .then(calorieTargets => response.status(200).send(calorieTargets))
      .catch(error => response.status(409));
  });

  app.post('/api/calorietargets', withAuth, (request, response) => {
    handlePostCalorieTarget(request)
      .then(calorieTargets => response.status(201).send(calorieTargets))
      .catch(error => response.status(409));
  });
};
