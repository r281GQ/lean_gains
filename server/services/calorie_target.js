const mongoose = require('mongoose');
const moment = require('moment');
const _ = require('lodash');
const CalorieTarget = mongoose.model('CalorieTarget');

const { decorateWithUser } = require('./../utils/service');

const setStartDateAndIsLatest = (body, user) =>
  decorateWithUser('CalorieTarget')(
    _.extend({}, body, {
      startDate: moment().valueOf(),
      isLatest: true
    }),
    user
  );

const findFormerLatest = user =>
  CalorieTarget.find({ user }).sort({ startDate: -1 }).skip(1).limit(1);

const updateFormerLatest = target =>
  !target[0]
    ? 0
    : CalorieTarget.findOneAndUpdate(
        { _id: target[0]._id },
        { $set: { endDate: moment().valueOf(), isLatest: false } }
      );

const handleGetCalorieTarget = ({ user }) =>
  new Promise((resolve, reject) =>
    CalorieTarget.find({ user })
      .then(targets => resolve(targets))
      .catch(err => reject(error))
  );
const handlePostCalorieTarget = ({ user, body }) =>
  new Promise((resolve, reject) =>
    setStartDateAndIsLatest(body, user)
      .save()
      .then(() => findFormerLatest(user))
      .then(target => updateFormerLatest(target))
      .then(() => CalorieTarget.find({ user }))
      .then(targets => resolve(targets))
      .catch(error => reject(error))
  );

module.exports = {
  handleGetCalorieTarget,
  handlePostCalorieTarget
};
