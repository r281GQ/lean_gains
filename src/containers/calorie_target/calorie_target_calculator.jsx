import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { reduxForm, formValueSelector, initialize } from 'redux-form/immutable';
import { connect } from 'react-redux';
import * as _ from 'lodash';
import { Map } from 'immutable';

import { createKcalTarget } from './../../store/actionCreators/user_details_action_creators';

import calorieTarget from './../../store/selectors/calorie_target';

import BMRCalculationSelector from './../../components/calorie_target/bmr_calculation_selector';
import BodyFatField from './../../components/calorie_target/body_fat_field';
import ActivityLevelSelecor from './../../components/calorie_target/activity_level_selector';
import CalorieSplitSelector from './../../components/calorie_target/calorie_split_selector';
import CustomCalorieField from './../../components/calorie_target/custom_calorie_field';
import ProteinField from './../../components/calorie_target/protein_field';
import FatMethodSelector from './../../components/calorie_target/fat_mathod_selector';
import FatSelector from './../../components/calorie_target/fat_selector';
import CenteredSubmitButton from './../../components/centered_submit_button';

class CalorieTargetCalculator extends Component {
  _adjustCaloriePercentage(props, nextProps) {
    const { calorieSplit } = nextProps;
    const { change } = props;
    if (calorieSplit === 'slowbulk') {
      change('restDay', -10);
      change('trainingDay', 30);
    }
    if (calorieSplit === 'cut') {
      change('restDay', -30);
      change('trainingDay', 10);
    }
    if (calorieSplit === 'recomp') {
      change('restDay', -20);
      change('trainingDay', 20);
    }
  }

  _adjustFatRatio(
    {
      change,
      restFatGrams,
      restFatPercentage,
      trainingFatGrams,
      trainingFatPercentage
    },
    { calorieTarget }
  ) {
    const maxRestGram = calorieTarget.getIn(['max', 'restGram']);
    const maxTrainingGram = calorieTarget.getIn(['max', 'trainingGram']);
    const maxRestPercentage = calorieTarget.getIn(['max', 'restPercentage']);
    const maxTrainingPercentage = calorieTarget.getIn([
      'max',
      'trainingPercentage'
    ]);

    if (restFatGrams > maxRestGram)
      change('restFatGrams', _.floor(maxRestGram));

    if (trainingFatGrams > maxTrainingGram)
      change('trainingFatGrams', _.floor(maxTrainingGram));

    if (restFatPercentage > maxRestPercentage)
      change('restFatPercentage', _.floor(maxRestPercentage));
    if (trainingFatPercentage > maxTrainingPercentage)
      change('trainingFatPercentage', _.floor(maxTrainingPercentage));
  }

  shouldComponentUpdate({
    calorieTarget,
    calorieSplit,
    restDay,
    trainingDay,
    activity,
    fatMethod,
    bmrCalculationMethod
  }) {
    const {
      restFatGrams,
      restFatPercentage,
      trainingFatGrams,
      trainingFatPercentage
    } = this.props;

    const maxRestGram = calorieTarget.getIn(['max', 'restGram']);
    const maxTrainingGram = calorieTarget.getIn(['max', 'trainingGram']);
    const maxRestPercentage = calorieTarget.getIn(['max', 'restPercentage']);
    const maxTrainingPercentage = calorieTarget.getIn([
      'max',
      'trainingPercentage'
    ]);

    const update =
      bmrCalculationMethod !== this.props.bmrCalculationMethod ||
      fatMethod !== this.props.fatMethod ||
      activity !== this.props.activity ||
      calorieSplit !== this.props.calorieSplit ||
      restDay !== this.props.restDay ||
      trainingDay !== this.props.trainingDay ||
      typeof restFatGrams === 'undefined' ||
      typeof this.props.restDay === 'undefined' ||
      restFatGrams > maxRestGram ||
      trainingFatGrams > maxTrainingGram ||
      restFatPercentage > maxRestPercentage ||
      trainingFatPercentage > maxTrainingPercentage;
    return update;
  }

  componentDidMount() {
    this.props.initializeForm(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this._adjustCaloriePercentage(this.props, nextProps);
    this._adjustFatRatio(this.props, nextProps);
  }

  render() {
    const {
      fatMethod,
      calorieSplit,
      handleSubmit,
      bmrCalculationMethod,
      createCalorieTarget,
      calorieTarget
    } = this.props;
    return (
      <div>
        <form
          onSubmit={handleSubmit(() =>
            createCalorieTarget(calorieTarget.get('finalValues').toJS())
          )}
        >
          <BMRCalculationSelector />
          <BodyFatField bmrCalculationMethod={bmrCalculationMethod} />
          <ActivityLevelSelecor
            normalize={value =>
              typeof value !== 'number' ? Number.parseFloat(value) : value}
          />
          <CalorieSplitSelector />
          <CustomCalorieField
            calorieSplit={calorieSplit}
            minRest={calorieTarget.toJS().minCalorie}
            minTraining={calorieTarget.toJS().minCalorie}
            normalize={value =>
              typeof value !== 'number' ? Number.parseFloat(value) : value}
          />
          <ProteinField />
          <FatMethodSelector />
          <FatSelector
            fatMethod={fatMethod}
            maxRestFatGrams={calorieTarget.getIn(['max', 'restGram'])}
            maxRestFatPercentage={calorieTarget.getIn([
              'max',
              'restPercentage'
            ])}
            maxTrainingFatGrams={calorieTarget.getIn(['max', 'trainingGram'])}
            maxTrainingFatPercentage={calorieTarget.getIn([
              'max',
              'trainingPercentage'
            ])}
          />
          <CenteredSubmitButton label="Create calorie target" />
        </form>
      </div>
    );
  }
}

CalorieTargetCalculator.propTypes = {
  activity: PropTypes.number,
  restDay: PropTypes.number,
  trainingDay: PropTypes.number,
  restFatGrams: PropTypes.number,
  restFatPercentage: PropTypes.number,
  trainingFatGrams: PropTypes.number,
  trainingFatPercentage: PropTypes.number,
  initializeForm: PropTypes.func,
  fatMethod: PropTypes.string,
  calorieSplit: PropTypes.string,
  bmrCalculationMethod: PropTypes.string,
  createCalorieTarget: PropTypes.func,
  calorieTarget: ImmutablePropTypes.mapContains({
    bodyFat: PropTypes.number,
    finalValues: ImmutablePropTypes.mapContains({
      rest: ImmutablePropTypes.mapContains({
        calorie: PropTypes.number,
        carbohydrate: PropTypes.number,
        fat: PropTypes.number,
        protein: PropTypes.number
      }),
      training: ImmutablePropTypes.mapContains({
        calorie: PropTypes.number,
        carbohydrate: PropTypes.number,
        fat: PropTypes.number,
        protein: PropTypes.number
      })
    }),
    minCalorie: PropTypes.number,
    max: ImmutablePropTypes.mapContains({
      restGram: PropTypes.number,
      restPercentage: PropTypes.number,
      trainingGram: PropTypes.number,
      trainingPercentage: PropTypes.number
    })
  })
};

// CalorieTargetCalculator.propTypes = {
//   activity: PropTypes.string,
//   restDay: PropTypes.number,
//   trainingDay: PropTypes.number,
//   restFatGrams: PropTypes.number,
//   restFatPercentage: PropTypes.number,
//   trainingFatGrams: PropTypes.number,
//   trainingFatPercentage: PropTypes.number,
//   initializeForm: PropTypes.func,
//   fatMethod: PropTypes.string,
//   calorieSplit: PropTypes.string,
//   bmrCalculationMethod: PropTypes.string,
//   createCalorieTarget: PropTypes.func,
//   calorieTarget: PropTypes.shape({
//     bodyFat: PropTypes.number,
//     finalValues: PropTypes.shape({
//       rest: PropTypes.shape({
//         calorie: PropTypes.number,
//         carbohydrate: PropTypes.number,
//         fat: PropTypes.number,
//         protein: PropTypes.number,
//       }),
//       training: PropTypes.shape({
//         calorie: PropTypes.number,
//         carbohydrate: PropTypes.number,
//         fat: PropTypes.number,
//         protein: PropTypes.number,
//       }),
//     }),
//     minCalorie: PropTypes.number,
//     max: PropTypes.shape({
//       restGram: PropTypes.number,
//       restPercentage: PropTypes.number,
//       trainingGram: PropTypes.number,
//       trainingPercentage: PropTypes.number,
//     }),
//   }),
// };

const selector = formValueSelector('calorie-target');

/*eslint no-class-assign: "off"*/
CalorieTargetCalculator = connect(state => ({
  activity: selector(state, 'activity'),
  fatMethod: selector(state, 'fatMethod'),
  calorieSplit: selector(state, 'calorieSplit'),
  bmrCalculationMethod: selector(state, 'bmrCalculationMethod'),
  trainingFatGrams: selector(state, 'trainingFatGrams'),
  restFatGrams: selector(state, 'restFatGrams'),
  trainingFatPercentage: selector(state, 'trainingFatPercentage'),
  restFatPercentage: selector(state, 'restFatPercentage'),
  restDay: selector(state, 'restDay'),
  trainingDay: selector(state, 'trainingDay')
}))(CalorieTargetCalculator);

const mapStateToProps = state => {
  return {
    calorieTarget: calorieTarget(state)
  };
};

const mapDispatchToProps = dispatch => {
  return {
    initializeForm: ({ calorieTarget }) =>
      dispatch(
        initialize(
          'calorie-target',
          Map().withMutations(map =>
            map
              .set('bodyFat', calorieTarget.get('bodyFat'))
              .set('restDay', -20)
              .set('trainingDay', 20)
              .set('restFatGrams', 0)
              .set('trainingFatGrams', 0)
              .set('restFatPercentage', 0)
              .set('trainingFatPercentage', 0)
              .set('fatMethod', 'grams')
              .set('activity', 1.2)
              .set('protein', 2)
              .set('calorieSplit', 'recomp')
              .set('bmrCalculationMethod', 'harris-benedict')
          )
        )
      ),
    createCalorieTarget: calorieTarget =>
      dispatch(createKcalTarget(calorieTarget))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(
  reduxForm({ form: 'calorie-target' })(CalorieTargetCalculator)
);
