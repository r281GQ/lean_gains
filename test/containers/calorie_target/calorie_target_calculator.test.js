import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { fromJS } from 'immutable';
import { PureCalorieTargetCalculator } from './../../../src/containers/calorie_target/calorie_target_calculator';

const f = fromJS({
  finalValues: {}
});

const nextProps = fromJS({

    max: {
      restGram: 0,
      trainingGram: 0,
      restPercentage: 0,
      trainingPercentage: 0
    }

});

describe('CalorieTargetCalculator test', () => {
  it('should render', () => {
    const component = shallow(
      <PureCalorieTargetCalculator
        calorieTarget={f}
        handleSubmit={() => () => undefined}
        createCalorieTarget={() => undefined}
        initializeForm={() => undefined}
      />
    );

    const instance = component.instance();

    const v = instance.shouldComponentUpdate({calorieTarget:nextProps});

    // console.log(v);

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });



  it('should shouldComponentUpdate return false when all the important properties are met', () => {
    const component = shallow(
      <PureCalorieTargetCalculator
        bmrCalculationMethod ='method'
        fatMethod = 'method'
        activity ={1}
        calorieSplit= 'split'
        restDay= {20}
        trainingDay= {20}
        restFatGrams = {0}
        restFatPercentage = {0}
        trainingFatGrams = {0}
        trainingFatPercentage ={0}
        calorieTarget={f}
        handleSubmit={() => () => undefined}
        createCalorieTarget={() => undefined}
        initializeForm={() => undefined}
      />
    );

    const g = fromJS({

        max: {
          restGram: 10,
          trainingGram: 10,
          restPercentage: 10,
          trainingPercentage: 10
        }

    });


    const next = {
      calorieTarget: g,
      bmrCalculationMethod : 'method',
      fatMethod: 'method',
      activity: 1,
      calorieSplit: 'split',
      restDay: 20,
      trainingDay: 20

    }


    const instance = component.instance();

    expect(instance.shouldComponentUpdate(next)).toBe(false);

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should shouldComponentUpdate return false when all the important properties are met', () => {
    const component = shallow(
      <PureCalorieTargetCalculator
        bmrCalculationMethod ='method'
        fatMethod = 'method'
        activity ={1}
        calorieSplit= 'split'
        restDay= {20}
        trainingDay= {20}
        restFatGrams = {0}
        restFatPercentage = {0}
        trainingFatGrams = {0}
        trainingFatPercentage ={0}
        calorieTarget={f}
        handleSubmit={() => () => undefined}
        createCalorieTarget={() => undefined}
        initializeForm={() => undefined}
      />
    );

    const g = fromJS({

        max: {
          restGram: 10,
          trainingGram: 10,
          restPercentage: 10,
          trainingPercentage: 10
        }

    });


    const next = {
      calorieTarget: g,
      bmrCalculationMethod : 'method',
      fatMethod: 'method',
      activity: 1,
      calorieSplit: 'slowbulk',
      restDay: 20,
      trainingDay: 20

    }


    const instance = component.instance();

    const changeSpy = sinon.spy()

    instance._adjustCaloriePercentage({change:changeSpy},next);


    expect(changeSpy.calledWith('restDay', -10)).toBe(true);
    expect(changeSpy.calledWith('trainingDay', 30)).toBe(true);

    instance._adjustCaloriePercentage({change:changeSpy},{calorieSplit: 'cut'});


    expect(changeSpy.calledWith('restDay', -30)).toBe(true);
    expect(changeSpy.calledWith('trainingDay', 10)).toBe(true);


    instance._adjustCaloriePercentage({change:changeSpy},{calorieSplit: 'recomp'});


    expect(changeSpy.calledWith('restDay', -20)).toBe(true);
    expect(changeSpy.calledWith('trainingDay', 20)).toBe(true);


    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should shouldComponentUpdate return false when all the important properties are met', () => {
    const component = shallow(
      <PureCalorieTargetCalculator
        bmrCalculationMethod ='method'
        fatMethod = 'method'
        activity ={1}
        calorieSplit= 'split'
        restDay= {20}
        trainingDay= {20}
        restFatGrams = {0}
        restFatPercentage = {0}
        trainingFatGrams = {0}
        trainingFatPercentage ={0}
        calorieTarget={f}
        handleSubmit={() => () => undefined}
        createCalorieTarget={() => undefined}
        initializeForm={() => undefined}
      />
    );

    const g = fromJS({

        max: {
          restGram: 10,
          trainingGram: 10,
          restPercentage: 10,
          trainingPercentage: 10
        }

    });


    const actual =  {
      restFatGrams: 20,
      trainingFatGrams: 20,
      restFatPercentage: 20,
      trainingFatPercentage: 20
    }

    const next = {
      calorieTarget: g,
      bmrCalculationMethod : 'method',
      fatMethod: 'method',
      activity: 1,
      calorieSplit: 'slowbulk',
      restDay: 20,
      trainingDay: 20

    }


    const instance = component.instance();

    const changeSpy = sinon.spy()

    instance._adjustFatRatio({change:changeSpy, ...actual },next);

    // console.log(changeSpy);
    expect(changeSpy.calledWith('restFatGrams', 10)).toBe(true);
    expect(changeSpy.calledWith('trainingFatGrams', 10)).toBe(true);
    expect(changeSpy.calledWith('restFatPercentage', 10)).toBe(true);
    expect(changeSpy.calledWith('trainingFatPercentage', 10)).toBe(true);
    // expect(changeSpy.calledWith('trainingDay', 30)).toBe(true);

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
