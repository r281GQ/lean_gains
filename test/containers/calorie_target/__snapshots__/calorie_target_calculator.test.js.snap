// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalorieTargetCalculator test should render 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <BMRCalculationSelector />
    <BodyFatField />
    <ActivityLevelSelecor
      normalize={[Function]}
    />
    <CalorieSplitSelector />
    <CustomCalorieField
      normalize={[Function]}
    />
    <ProteinField />
    <FatMethodSelector />
    <FatSelector />
    <CenteredSubmitButton
      label="Create calorie target"
    />
  </form>
</div>
`;

exports[`CalorieTargetCalculator test should shouldComponentUpdate return false when all the important properties are met 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <BMRCalculationSelector />
    <BodyFatField
      bmrCalculationMethod="method"
    />
    <ActivityLevelSelecor
      normalize={[Function]}
    />
    <CalorieSplitSelector />
    <CustomCalorieField
      calorieSplit="split"
      normalize={[Function]}
    />
    <ProteinField />
    <FatMethodSelector />
    <FatSelector
      fatMethod="method"
    />
    <CenteredSubmitButton
      label="Create calorie target"
    />
  </form>
</div>
`;

exports[`CalorieTargetCalculator test should shouldComponentUpdate return false when all the important properties are met 2`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <BMRCalculationSelector />
    <BodyFatField
      bmrCalculationMethod="method"
    />
    <ActivityLevelSelecor
      normalize={[Function]}
    />
    <CalorieSplitSelector />
    <CustomCalorieField
      calorieSplit="split"
      normalize={[Function]}
    />
    <ProteinField />
    <FatMethodSelector />
    <FatSelector
      fatMethod="method"
    />
    <CenteredSubmitButton
      label="Create calorie target"
    />
  </form>
</div>
`;

exports[`CalorieTargetCalculator test should shouldComponentUpdate return false when all the important properties are met 3`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <BMRCalculationSelector />
    <BodyFatField
      bmrCalculationMethod="method"
    />
    <ActivityLevelSelecor
      normalize={[Function]}
    />
    <CalorieSplitSelector />
    <CustomCalorieField
      calorieSplit="split"
      normalize={[Function]}
    />
    <ProteinField />
    <FatMethodSelector />
    <FatSelector
      fatMethod="method"
    />
    <CenteredSubmitButton
      label="Create calorie target"
    />
  </form>
</div>
`;
